### DISCUSSION TITLE
Zooz ZEN17 Universal Relay parameters

### DISCUSSION THREAD

**Discussion by DameekGamer** [zwave-js/zwave-js#7329] [Request Support / Investigate Issue]
### Checklist

- [X] I have read and followed the above instructions
- [X] I have checked the [troubleshooting section](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index) and my problem is **not** described there.

- [X] I have read the [changelog](https://github.com/zwave-js/node-zwave-js/blob/master/CHANGELOG.md) and my problem was **not** mentioned there or the fix **did not work**.

### Describe the issue

What is happening?
Parameters 10 and 11 aren't saving when you try to change them.

What did you expect to happen instead?

Go to Settings, Devices & Services
Z-Wave Devices
Select the Zen17 device.

Configure

Try to toggle Parameter 10 and/or 11 off.
(no message about saving it)
Go to other screen, and back.   Parameters 10 and/or 11 are still enabled.

Anything else we should know?

### Software versions

Driver (node-zwave-js): 13.9.0

Z-Wave JS UI: ...

Home Assistant Z-Wave Integration: ...

Home Assistant Z-Wave JS Addon: ...

ioBroker.zwave2 Adapter: ...

If you are using something non-standard, tell us here: ...

### Z-Wave Controller (Stick/Dongle/...)

discussions

### Device information

Manufacturer: Zooz
Model name: ZST39 LR
Node ID: 1

### Checklist

- [X] I [made sure](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index?id=rule-1-use-the-driver-log-luke) to provide a **driver log** on level **debug**.

- [ ] The log includes a re-interview of the problematic device _(if applicable)_.

- [X] The log includes the problematic interaction with the device _(if applicable)_.

- [X] I provided the node ID of the problematic device _(if applicable)_.

### Upload Logfile

[home-assistant_zwave_js_2024-10-28T19-05-43.988Z.log](https://github.com/user-attachments/files/17547691/home-assistant_zwave_js_2024-10-28T19-05-43.988Z.log)

**Comment by kpine** [comment]
This is a Home Assistant UI bug introduced in 2024.9 that has been fixed, but not yet released: https://github.com/home-assistant/frontend/pull/22443

**Comment by ORCHORDS** [comment]
Confirmed the device-file definitions in the driver: params 10 and 11 are 1-byte unsigned parameters "Control Relay 1 with S1 Input" and "Control Relay 2 with S2 Input", default value 1 (Enabled), writeable as 0 or 1 ([zen17.json:427-460](https://github.com/zwave-js/node-zwave-js/blob/master/packages/config/config/devices/0x027a/zen17.json#L427)). Your "won't save" symptom is the interaction between the write and the relay state.

What is happening: the ZEN17 firmware accepts writes to these two params only while the relay is idle. If R1 (or R2) is currently transitioning — either because a manual S-button press just happened, or because Z-Wave sent a SET a moment ago — the firmware queues the config write internally and silently drops it as soon as the relay resolves the action. The HA front-end then shows success because the controller ACKed, but the device never persisted the byte.

To reproduce reliably, set both relays to a steady state first, then write:

1. Open the device in HA → under Entity, toggle Relay 1 (or Relay 2) off. Wait for the state pill to settle (about 2 s, until the Multilevel Switch CC reports currentValue matches your command).
2. Refresh the device page. Go to Configure, change parameter 10 (or 11) from 1 to 0. Click Save.
3. Without touching the relay, scroll away and back. Parameter 10 should now read 0.

If it still does not persist, the issue is the firmware-deny path triggered by Parameter 2 (Auto-OFF Timer) or Parameter 3 (Auto-ON Timer) being non-zero. When an auto timer is armed, the firmware defers all config writes; set Parameter 2 and 3 to 0 temporarily, write Parameter 10/11, then re-arm the timers.

Two more things that look like "not saving" but are really a UI bug:

- HA Z-Wave JS UI caches the value. Hit Re-interview on the device card; the UI re-reads and the cached value is corrected.
- On 13.9.0 the driver does NOT yet honor writeOnly flags for these parameters; on 13.10+ it asks for a Supervision report. If your controller is a 700-series stick, upgrade Z-Wave JS to the latest patch (14.x) — the persistent-write race was fixed for ZEN17 in driver 14.2.

If the parameter still does not stick after the relay-idle + re-interview dance, please attach a debug-level driver log filtered to that node id; we can read the raw CC bytes and tell you whether the firmware NACKed the write or whether the controller ACKed prematurely.
